src/OpenTelemetry.AutoInstrumentation/Diagnostics/SdkSelfDiagnosticsEventListener.cs

```diff
@@ -31,13 +31,19 @@ internal class SdkSelfDiagnosticsEventListener : EventListener
 {
     private const string EventSourceNamePrefix = "OpenTelemetry-";
 
-    private static readonly ILogger Log = OtelLogging.GetLogger();
     private readonly object lockObj = new();
     private readonly EventLevel logLevel;
+    private readonly ILogger log;
     private readonly List<EventSource> eventSourcesBeforeConstructor = new();
 
     public SdkSelfDiagnosticsEventListener(EventLevel eventLevel)
+        : this(eventLevel, OtelLogging.GetLogger())
     {
+    }
+
+    internal SdkSelfDiagnosticsEventListener(EventLevel eventLevel, ILogger logger)
+    {
```

Comment:
Can we have a single constructor (just this one)?

References:
- https://blogs.cuttingedge.it/steven/posts/2013/di-anti-pattern-multiple-constructors/
- https://blog.ploeh.dk/2011/05/30/DesignSmellDefaultConstructor/

Reply:
I don't have any strong opinion on this as we don't have DI anyway. Default constructor would have been more futureproof if `GetLogger` is changed to `GetLoggerFor` as passing the wrong instance. Currently, as the logger is static all over the app anyway, I'm fine with refactored approach also.